modules/core/src/main/scala/org/scalasteward/core/scalafix/Migration.scala

```diff
@@ -17,12 +17,54 @@
 package org.scalasteward.core.scalafix
 
 import org.scalasteward.core.data.{GroupId, Version}
+import cats.Eq
 import org.scalasteward.core.util.Nel
 import scala.util.matching.Regex
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import io.circe.Decoder._
+import cats.syntax.eq._
+import cats.instances.string._
+import cats.kernel.Hash
+import cats.instances.tuple._
+import cats.instances.int._
+import cats.instances.list._
 
 final case class Migration(
     groupId: GroupId,
     artifactIds: Nel[Regex],
     newVersion: Version,
     rewriteRules: Nel[String]
-)
+) {
+
+  override def hashCode: Int = Hash[Migration].hash(this)
```

Comment:
@manuelcueto It isn't obvious to me why you override `hashCode`, `equals`, and why there is a `Hash` instance. Where is this used?

Reply:
migration's default equals was failing for 2 migrations with the same stuff. This is due to 2 regex instances being not equal even if they were formed with the same string. I brought this in order to use a custom equals for Regex. But ideally i wouldn't want to do this at all. Do you have any suggestions?

Reply:
Would it make things simpler if we change the type of `artifactIds: Nel[Regex]` to `Nel[String]` and convert them to regexes later when we're trying to find suitable migrations for an update?

Reply:
Seems sensible